ovos_audio/playback.py

```diff
@@ -0,0 +1,272 @@
+import random
+import threading
+from ovos_audio.transformers import TTSTransformersService
+from ovos_bus_client.message import Message
+from ovos_plugin_manager.templates.tts import TTS
+from ovos_utils.log import LOG
+from ovos_utils.sound import play_audio
+from queue import Empty
+from threading import Thread
+from time import time, sleep
+
+
+class PlaybackThread(Thread):
+    """Thread class for playing back tts audio and sending
+    viseme data to enclosure.
+    """
+
+    def __init__(self, queue=TTS.queue, bus=None):
+        super(PlaybackThread, self).__init__()
+        self.queue = queue or TTS.queue
+        self._terminated = False
+        self._processing_queue = False
+        self._paused = False
+        self.enclosure = None
+        self.p = None
+        self._tts = []
+        self.bus = bus or None
+        self._now_playing = None
+        self.active_tts = None
+        self._started = threading.Event()
+        self.tts_transform = TTSTransformersService(self.bus)
+
+    @property
+    def is_running(self):
+        return self._started.is_set() and not self._terminated
+
+    def activate_tts(self, tts_id):
+        self.active_tts = tts_id
+        tts = self.get_attached_tts()
+        if tts:
+            tts.begin_audio()
+
+    def deactivate_tts(self):
+        if self.active_tts:
+            tts = self.get_attached_tts()
+            if tts:
+                tts.end_audio()
+        self.active_tts = None
+
+    def init(self, tts):
+        """DEPRECATED! Init the TTS Playback thread."""
```

Comment:
No log message for this yet?

Reply:
not touching old code in this file in this PR, it was just moved from OPM to here.

the only addition is the step of tts_transformers, cleaning up the old code will probably need more considerations for backwards compat